dss/storage/bundles.py

```diff
@@ -22,44 +22,66 @@
     f"^{BUNDLE_PREFIX}/({UUID_PATTERN})(?:\.(" + VERSION_PATTERN + "))?\." + TOMBSTONE_SUFFIX + "$")
 # matches all bundle objects
 DSS_OBJECT_NAME_REGEX = re.compile(
-    f"^{BUNDLE_PREFIX}/({UUID_PATTERN})(?:\.({VERSION_PATTERN}))?(?:\.{TOMBSTONE_SUFFIX})?$")
+    f"^({BUNDLE_PREFIX}|{FILE_PREFIX})/({UUID_PATTERN})(?:\.({VERSION_PATTERN}))?(\.{TOMBSTONE_SUFFIX})?$")
 
 
-def bundle_key_to_bundle_fqid(bundle_key: str) -> str:
-    match = DSS_OBJECT_NAME_REGEX.match(bundle_key)
-    uuid, version = match.groups() if match else (None, None)
-    if uuid and version:
-        return format_bundle_fqid(uuid, version)
-    else:
-        raise ValueError(f"Object name does not contain a valid bundle identifier: {bundle_key}")
+class ObjectIdentifier(namedtuple('ObjectIdentifier', 'uuid version')):
 
+    @classmethod
+    def from_key(cls, bundle_key: str):
+        match = DSS_OBJECT_NAME_REGEX.match(bundle_key)
+        object_type, uuid, version, tombstone_suffix = match.groups() if match else (None, None, None, None)
```

Comment:
We should, consider—in a different refactoring—to move from `str` to `uuid` for the type of the `uuid` member. UUIDs are not hexadecimal strings by nature. The hex str representation is ambiguous wrt to case. I've seen crazy bugs cause by that.

Reply:
💯